README.md
# Search Bar/Box UI 

Simple search box UI with clear text icon.
Highly customizable `search` bar.


| Android |
| ------------- |
| <img src="searchbar.gif" width="300" height="600">|



## Getting Started

Here's a simple example of `Search Bar`'s:

    import 'package:flutter/material.dart';
    import 'package:simple_textfield_search_bar/searchBox.dart';



    class Search extends StatefulWidget {
    @override
    _SearchState createState() => _SearchState();
    }

    class _SearchState extends State<Search> {
    @override
    Widget build(BuildContext context) {
        double w = MediaQuery.of(context).size.width;
        return Scaffold(
        body: InkWell(
            onTap: () {
            FocusScope.of(context).unfocus();
            },
            child: Center(
            child: Container(
                margin: EdgeInsets.only(
                left: w / 35,
                right: w / 35,
                ),
                child: SearchBar(
                    buttonStyle: TextStyle(
                        color: Colors.white,
                        fontSize: 16,
                        fontFamily: 'Gilroy-ExtraBold',
                        fontWeight: FontWeight.w500),
                    hintText: 'Search',
                    icon: Icons.cancel,
                    onSubmitted: (value) {
                    print(value);
                    },
                    radius: 10.5,
                    title: "heel",
                    bgcolor: Color(0xFF905BE0),
                    height: MediaQuery.of(context).size.height,
                    iconColor: Colors.white),
            ),
            ),
        ),
        );
    }
    }

## Parameters

Here's a list of all `SearchAppBar` parameters and what they do:

`height` : Height of the search bar.

`bgcolor` : Background Color of the search bar.

`radius` : Circular radius of the search bar.

`buttonStyle` : Text Style 

`hintText` : Hint Text

`icon` : Suffix Icon for Text field

`iconColor` : Suffix Icon Color for Text field

`onSubmitted` : onSubmitted Function 










CHANGELOG.md
## 0.0.1

* Initial release for Android and iOS.

lib/main.dart
import 'package:flutter/material.dart';

import 'example.dart';

void main() {
  runApp(new MaterialApp(home: Search()));
}

class SearchBar extends StatefulWidget {
  SearchBar(
      {Key key,
      this.title,
      this.buttonStyle,
      this.onSubmitted,
      this.radius,
      this.hintText,
      this.icon,
      this.bgcolor,
      this.height,
      this.iconColor})
      : super(key: key);
  final TextStyle buttonStyle;
  final String title;
  final Function onSubmitted;
  final radius;
  final String hintText;
  final IconData icon;
  final Color bgcolor;
  final height;
  final Color iconColor;

  @override
  _SearchBarState createState() => _SearchBarState();
}

class _SearchBarState extends State<SearchBar> {
  FocusNode focusNode;
  bool clear = true;
  TextEditingController search = TextEditingController();

  @override
  Widget build(BuildContext context) {
    return Container(
      height: widget.height / 15,
      decoration: BoxDecoration(
        borderRadius: BorderRadius.circular(widget.radius.toDouble()),
        color: widget.bgcolor,
      ),
      child: Row(
        children: [
          Expanded(
            flex: 2,
            child: TextField(
                focusNode: focusNode,
                controller: search,
                style: widget.buttonStyle,
                decoration: InputDecoration(
                  border: InputBorder.none,
                  focusedBorder: InputBorder.none,
                  enabledBorder: InputBorder.none,
                  errorBorder: InputBorder.none,
                  disabledBorder: InputBorder.none,
                  hintText: widget.hintText,
                  hintStyle: widget.buttonStyle,
                  prefixText: '  ',
                ),
                onTap: () {
                  setState(() {
                    clear = true;
                  });
                },
                onSubmitted: (v) => widget.onSubmitted(v)),
          ),
          clear
              ? IconButton(
                  icon: Icon(widget.icon,
                      color: clear ? widget.iconColor : Colors.black),
                  onPressed: () {
                    search.clear();
                    setState(() {
                      clear = false;
                    });
                  },
                )
              : Icon(
                  Icons.ac_unit,
                  size: .10,
                )
        ],
      ),
    );
  }
}


lib/searchBox.dart
import 'package:flutter/material.dart';

class SearchBar extends StatefulWidget {
  SearchBar(
      {Key key,
      this.title,
      this.buttonStyle,
      this.onSubmitted,
      this.radius,
      this.hintText,
      this.icon,
      this.bgcolor,
      this.height,
      this.iconColor})
      : super(key: key);
  final TextStyle buttonStyle;
  final String title;
  final Function onSubmitted;
  final radius;
  final String hintText;
  final IconData icon;
  final Color bgcolor;
  final height;
  final Color iconColor;

  @override
  _SearchBarState createState() => _SearchBarState();
}

class _SearchBarState extends State<SearchBar> {
  FocusNode focusNode;
  bool clear = true;
  TextEditingController search = TextEditingController();

  @override
  Widget build(BuildContext context) {
    return Container(
      height: widget.height / 15,
      decoration: BoxDecoration(
        borderRadius: BorderRadius.circular(widget.radius.toDouble()),
        color: widget.bgcolor,
      ),
      child: Row(
        children: [
          Expanded(
            flex: 2,
            child: TextField(
                focusNode: focusNode,
                controller: search,
                style: widget.buttonStyle,
                decoration: InputDecoration(
                  border: InputBorder.none,
                  focusedBorder: InputBorder.none,
                  enabledBorder: InputBorder.none,
                  errorBorder: InputBorder.none,
                  disabledBorder: InputBorder.none,
                  hintText: widget.hintText,
                  hintStyle: widget.buttonStyle,
                  prefixText: '  ',
                ),
                onTap: () {
                  setState(() {
                    clear = true;
                  });
                },
                onSubmitted: (v) => widget.onSubmitted(v)),
          ),
          clear
              ? IconButton(
                  icon: Icon(widget.icon,
                      color: clear ? widget.iconColor : Colors.black),
                  onPressed: () {
                    search.clear();
                    setState(() {
                      clear = false;
                    });
                  },
                )
              : Icon(
                  Icons.ac_unit,
                  size: .10,
                )
        ],
      ),
    );
  }
}


lib/example/example.dart
import 'package:flutter/material.dart';

import '../searchBox.dart';

void main() {
  runApp(new MaterialApp(home: Search()));
}

class Search extends StatefulWidget {
  @override
  _SearchState createState() => _SearchState();
}

class _SearchState extends State<Search> {
  @override
  Widget build(BuildContext context) {
    double w = MediaQuery.of(context).size.width;
    return Scaffold(
      body: InkWell(
        onTap: () {
          FocusScope.of(context).unfocus();
        },
        child: Center(
          child: Container(
            margin: EdgeInsets.only(
              left: w / 35,
              right: w / 35,
            ),
            child: SearchBar(
                buttonStyle: TextStyle(
                    color: Colors.white,
                    fontSize: 16,
                    fontFamily: 'Gilroy-ExtraBold',
                    fontWeight: FontWeight.w500),
                hintText: 'Search',
                icon: Icons.cancel,
                onSubmitted: (value) {
                  print(value);
                },
                radius: 10.5,
                title: "heel",
                bgcolor: Color(0xFF905BE0),
                height: MediaQuery.of(context).size.height,
                iconColor: Colors.white),
          ),
        ),
      ),
    );
  }
}
